tests/Microsoft.Bot.Builder.Tests/SlotTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Bot.Builder.Tests
+{
+    [TestClass]
+    [TestCategory("State")]
+    public class SlotTests
+    {
```

Comment:
Instead of all of these being prefixed with `Ctor_` I would recommend following the pattern of introducing a nested class for the method under test... in this case `public class ConstructorTests` and then putting the test methods inside of there. This works nicely with test runners such as VS's (finally) that understand the hierarchial test class pattern. This also means you move the `TestClass`/`TestCategory` attributes from the `SlotTests` parent class and put it on the nested classes instead... this is ceremony overhead of MSTest, XUnit you wouldn't have to apply an attribute the classes at all.

NOTE: this suggestion applies to all the tests in this PR, but I'm only writing this one comment.